src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -195,4 +265,28 @@ private function doSmallFileUpload(array $options, string $bucket, string $key,
             'Body' => $object,
         ]));
     }
+
+    private function doMultipartCopy(string $bucket, string $key, string $uploadId, int $partNumber, string $copySource, int $startByte, int $endByte): CompletedPart
+    {
+        try {
+            $response = $this->uploadPartCopy(
```

Comment:
this operation requires the (not yet release) next version of S3.
You need to either upgrade composer.json `"async-aws/s3": "^2.1"`

Reply:
https://github.com/async-aws/aws/pull/1591
First one